backend/tracim_backend/views/core_api/schemas.py

```diff
@@ -158,14 +158,30 @@ def _deserialize(self, value: typing.Any, *arg: typing.Any, **kwargs: typing.Any
                 )
             )
 
-    def _serialize(self, value: Enum, *arg: typing.Any, **kwargs: typing.Any) -> typing.Any:
-        if value not in self._enum:
+    def _serialize(
+        self, value: typing.Union[Enum, str], *arg: typing.Any, **kwargs: typing.Any
+    ) -> typing.Any:
+        # INFO - G.M - 2021-07-08 -
+        # EnumField support both Enum and valid string corresponding to enum
+        # Both are serialized enum value.
+        if isinstance(value, Enum):
+            if value not in (elem for elem in self._enum):
+                raise marshmallow.ValidationError(
+                    "'{}' is not a valid value for this field".format(value)
+                )
+            return value.value
+        elif isinstance(value, str):
+            try:
+                return self._enum(value).value
+            except Exception:
+                raise marshmallow.ValidationError(
+                    "'{}' is not a valid value for this field".format(value)
+                )
+        else:
             raise marshmallow.ValidationError(
-                "'{}' is not a valid value for this field".format(value)
+                "'{}' is not a valid type for this field".format(value)
             )
 
```

Comment:
I'm not sure this is the good solution.
There was 2 issues here:
1. "if value not in self._enum" is not valid in 3.8+ with value as string : the error that cause tests to fail.
2. a validation error here doesn't cause issue (hapic bug, see processor patch) and a part of our code is based on this.

The second issue has been found as it's seems unlogical to me that pushing string value here cause any issue in 3.6.

Another solution to fix this, is to be more strict and convert string to enum before this step. For now, the only known case where the issue exist (we put string in a EnumField), is advanced search where we don't do any change.